test/e2e/__init__.py

```diff
@@ -51,3 +54,25 @@ def create_sagemaker_resource(
     )
 
     return reference, spec, resource
+
+
+def wait_sagemaker_endpoint_status(
+    sagemaker_client,
+    endpoint_name,
+    expected_status: str,
+    wait_periods: int = 60,
+):
+    actual_status = None
+    for _ in range(wait_periods):
+        time.sleep(30)
+        actual_status = sagemaker_client.describe_endpoint(EndpointName=endpoint_name)[
```

Comment:
If we are moving this out to a common file, we should make this common across resources. One way could be to pass  `sagemaker_client.describe_endpoint(EndpointName=endpoint_name)` as a function parameter to the `wait_sagemaker_generic_status` function.

Reply:
I think we should make the change now, since I was reusing your logic from endpoint status wait across all other tests. That is a lot of duplication and this is how I was planning to fix it if I got the chance.

Reply:
got it, made it generic